vtkext/private/module/vtkF3DImguiActor.cxx

```diff
@@ -0,0 +1,311 @@
+      ImGui::DestroyContext();
+    }
+  }
+
+  void RenderDrawData(vtkOpenGLRenderWindow* renWin, ImDrawData* drawData)
+  {
+    vtkOpenGLState* state = renWin->GetState();
+
+    vtkOpenGLState::ScopedglScissor save_scissorbox(state);
+    vtkOpenGLState::ScopedglBlendFuncSeparate save_blendfunc(state);
+    vtkOpenGLState::ScopedglEnableDisable save_blend(state, GL_BLEND);
+    vtkOpenGLState::ScopedglEnableDisable save_cull(state, GL_CULL_FACE);
+    vtkOpenGLState::ScopedglEnableDisable save_depth(state, GL_DEPTH_TEST);
+    vtkOpenGLState::ScopedglEnableDisable save_stencil(state, GL_STENCIL_TEST);
+    vtkOpenGLState::ScopedglEnableDisable save_scissor(state, GL_SCISSOR_TEST);
+
+    // Change require OpenGL state for proper rendering
+    state->vtkglEnable(GL_BLEND);
+    state->vtkglBlendEquation(GL_FUNC_ADD);
+    state->vtkglBlendFuncSeparate(
+      GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA, GL_ONE, GL_ONE_MINUS_SRC_ALPHA);
+    state->vtkglDisable(GL_CULL_FACE);
+    state->vtkglDisable(GL_DEPTH_TEST);
+    state->vtkglDisable(GL_STENCIL_TEST);
+    state->vtkglEnable(GL_SCISSOR_TEST);
+
+    renWin->GetShaderCache()->ReadyShaderProgram(this->Program);
+
+    // Set scale/shift (Y is inverted in OpenGL)
+    float scale[2], shift[2];
+    scale[0] = 2.f / drawData->DisplaySize.x;
+    scale[1] = -2.f / drawData->DisplaySize.y;
+    shift[0] = -(2.f * drawData->DisplayPos.x + drawData->DisplaySize.x) / drawData->DisplaySize.x;
+    shift[1] = (2.f * drawData->DisplayPos.y + drawData->DisplaySize.y) / drawData->DisplaySize.y;
+
+    this->FontTexture->Activate();
+
+    this->Program->SetUniform2f("Scale", scale);
+    this->Program->SetUniform2f("Shift", shift);
+    this->Program->SetUniformi("Texture", this->FontTexture->GetTextureUnit());
+    
+    // Render the UI
+    this->VertexArray->Bind();
+    this->VertexBuffer->Bind();
+    this->IndexBuffer->Bind();
+
+    ImVec2 clipOff = drawData->DisplayPos;
+    ImVec2 clipScale = drawData->FramebufferScale;
+
+    for (int n = 0; n < drawData->CmdListsCount; n++)
+    {
+      const ImDrawList* cmdList = drawData->CmdLists[n];
+
+      this->VertexBuffer->Upload(
+        cmdList->VtxBuffer.Data, cmdList->VtxBuffer.Size, vtkOpenGLBufferObject::ArrayBuffer);
+      this->IndexBuffer->Upload(cmdList->IdxBuffer.Data, cmdList->IdxBuffer.Size,
+        vtkOpenGLBufferObject::ElementArrayBuffer);
+
+      for (int iCmd = 0; iCmd < cmdList->CmdBuffer.Size; iCmd++)
+      {
+        const ImDrawCmd* cmd = &cmdList->CmdBuffer[iCmd];
+
+        // Project scissor/clipping rectangles into framebuffer space
+        ImVec2 clipMin(
+          (cmd->ClipRect.x - clipOff.x) * clipScale.x, (cmd->ClipRect.y - clipOff.y) * clipScale.y);
+        ImVec2 clipMax(
+          (cmd->ClipRect.z - clipOff.x) * clipScale.x, (cmd->ClipRect.w - clipOff.y) * clipScale.y);
+        if (clipMax.x <= clipMin.x || clipMax.y <= clipMin.y)
+          continue;
```

Comment:
unreachable ?

Reply:
it can be reached if there's an error in imgui calls. I changed the condition logic to avoid a missing line in the coverage